subsys/net/ip/connection.c

```diff
@@ -752,15 +753,13 @@ enum net_verdict net_conn_input(enum net_ip_protocol proto, struct net_pkt *pkt)
 	}
 #endif
 
-	if (IS_ENABLED(CONFIG_NET_DEBUG_CONN)) {
-		u16_t chksum;
-
-		if (proto == IPPROTO_TCP) {
-			chksum = NET_TCP_HDR(pkt)->chksum;
-		} else {
-			chksum = NET_UDP_HDR(pkt)->chksum;
-		}
+	if (proto == IPPROTO_TCP) {
```

Comment:
Sure, but lets figure this out when we are adding new port based protocols to the stack.